DOC: Updating capitalization in folder doc/source/reference

cleconte987 opened the pull request:
- [X] xref #32550

cleconte987 committed:
Updating capitalization in folder doc/source/reference

WillAyd commented:
@datapythonista

datapythonista reviewed (approved):
Thanks for working on this @cleconte987, looks great.

What would be very useful is to add the files that are fixed to https://github.com/pandas-dev/pandas/blob/master/ci/code_checks.sh#L330, so no more errors are being introduced in those fails (the CI will fail and let us know if any PR introduces titles capitalized incorrectly).

If you can add it to this PR that would be great.

cleconte987 commented:
Ok I will work on that!


> Le 19 mars 2020 à 16:20, Marc Garcia <notifications@github.com> a écrit :
> 
> ﻿
> @datapythonista approved this pull request.
> 
> Thanks for working on this @cleconte987, looks great.
> 
> What would be very useful is to add the files that are fixed to https://github.com/pandas-dev/pandas/blob/master/ci/code_checks.sh#L330, so no more errors are being introduced in those fails (the CI will fail and let us know if any PR introduces titles capitalized incorrectly).
> 
> If you can add it to this PR that would be great.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.

cleconte987 committed:
Merge remote-tracking branch 'upstream_main_pandas/master' into changes_to_pandas_remote

cleconte987 committed:
Add folder doc/source/reference in ci code checking  in ci/code_checks.sh L 330

datapythonista commented:
Thanks @cleconte987, changes look great. But looks like there are still several errors in that directory. Most of them need to be added here: https://github.com/pandas-dev/pandas/blob/master/scripts/validate_rst_title_capitalization.py#L20

Do you mind adding them, and fixing the cases that need it, if any? Thanks!

cleconte987 committed:
Modification of validate_rst_title_capitalization.py script

cleconte987 committed:
Modify validate_rst_title_capitalization.py script to take into account words with multiple capital letters

cleconte987 committed:
remove stylistic errors in comment section L:77,78,79

cleconte987 committed:
Modify validate_rst_title_capitalization.py script

cleconte987 committed:
Merge remote-tracking branch 'upstream_main_pandas/master' into script_modification
Merging new changes from pandas-dev/pandas repository into script_modification to be up-to-date

cleconte987 committed:
modify script following tests

cleconte987 committed:
Modify second time following style checking

cleconte987 committed:
Update branch with pandas repository commits
Merge remote-tracking branch 'upstream_main_pandas/master' into script_modification

cleconte987 committed:
Commit to update capitalization in folder doc/source/reference and to add elements to exceptions's set in scripts/validate_rst_title_capitalization.py

cleconte987 committed:
Merge branches to update PR
Merge branch 'script_modification' into changes_to_pandas_remote

cleconte987 committed:
Merge remote-tracking branch 'upstream_main_pandas/master' into script_modification

cleconte987 committed:
Merge remote-tracking branch 'temporary_repo/changes_to_pandas_remote' into changes_to_pandas_remote

pep8speaks commented:
Hello @cleconte987! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`scripts/validate_rst_title_capitalization.py`](https://github.com/pandas-dev/pandas/blob/62d516413e697e1823c904be3023e4c1f553ddda/scripts/validate_rst_title_capitalization.py):

> [Line 18:3](https://github.com/pandas-dev/pandas/blob/62d516413e697e1823c904be3023e4c1f553ddda/scripts/validate_rst_title_capitalization.py#L18): [E227](https://duckduckgo.com/?q=pep8%20E227) missing whitespace around bitwise or shift operator
> [Line 18:7](https://github.com/pandas-dev/pandas/blob/62d516413e697e1823c904be3023e4c1f553ddda/scripts/validate_rst_title_capitalization.py#L18): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 19:3](https://github.com/pandas-dev/pandas/blob/62d516413e697e1823c904be3023e4c1f553ddda/scripts/validate_rst_title_capitalization.py#L19): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 19:5](https://github.com/pandas-dev/pandas/blob/62d516413e697e1823c904be3023e4c1f553ddda/scripts/validate_rst_title_capitalization.py#L19): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 19:7](https://github.com/pandas-dev/pandas/blob/62d516413e697e1823c904be3023e4c1f553ddda/scripts/validate_rst_title_capitalization.py#L19): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 21:7](https://github.com/pandas-dev/pandas/blob/62d516413e697e1823c904be3023e4c1f553ddda/scripts/validate_rst_title_capitalization.py#L21): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 21:23](https://github.com/pandas-dev/pandas/blob/62d516413e697e1823c904be3023e4c1f553ddda/scripts/validate_rst_title_capitalization.py#L21): [E226](https://duckduckgo.com/?q=pep8%20E226) missing whitespace around arithmetic operator
> [Line 107:1](https://github.com/pandas-dev/pandas/blob/62d516413e697e1823c904be3023e4c1f553ddda/scripts/validate_rst_title_capitalization.py#L107): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented
> [Line 107:3](https://github.com/pandas-dev/pandas/blob/62d516413e697e1823c904be3023e4c1f553ddda/scripts/validate_rst_title_capitalization.py#L107): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 107:5](https://github.com/pandas-dev/pandas/blob/62d516413e697e1823c904be3023e4c1f553ddda/scripts/validate_rst_title_capitalization.py#L107): [E227](https://duckduckgo.com/?q=pep8%20E227) missing whitespace around bitwise or shift operator
> [Line 108:1](https://github.com/pandas-dev/pandas/blob/62d516413e697e1823c904be3023e4c1f553ddda/scripts/validate_rst_title_capitalization.py#L108): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented
> [Line 108:3](https://github.com/pandas-dev/pandas/blob/62d516413e697e1823c904be3023e4c1f553ddda/scripts/validate_rst_title_capitalization.py#L108): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 108:5](https://github.com/pandas-dev/pandas/blob/62d516413e697e1823c904be3023e4c1f553ddda/scripts/validate_rst_title_capitalization.py#L108): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 108:7](https://github.com/pandas-dev/pandas/blob/62d516413e697e1823c904be3023e4c1f553ddda/scripts/validate_rst_title_capitalization.py#L108): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 110:1](https://github.com/pandas-dev/pandas/blob/62d516413e697e1823c904be3023e4c1f553ddda/scripts/validate_rst_title_capitalization.py#L110): [E122](https://duckduckgo.com/?q=pep8%20E122) continuation line missing indentation or outdented
> [Line 110:7](https://github.com/pandas-dev/pandas/blob/62d516413e697e1823c904be3023e4c1f553ddda/scripts/validate_rst_title_capitalization.py#L110): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 110:23](https://github.com/pandas-dev/pandas/blob/62d516413e697e1823c904be3023e4c1f553ddda/scripts/validate_rst_title_capitalization.py#L110): [E226](https://duckduckgo.com/?q=pep8%20E226) missing whitespace around arithmetic operator

cleconte987 committed:
Remove wrong lines of code

cleconte987 committed:
Formatting

datapythonista commented on doc/source/reference/series.rst:
```diff
@@ -110,7 +110,7 @@ Binary operator functions
    Series.dot
 
-Function application, groupby & window
+Function application, GroupBy & window
```
What's the reason for this one? Just my opinion, but I think this one was correct, and we don't want to capitalize `groupy` to `GroupBy`.

cleconte987 replied:
I might have wrongly seen it with capital letters or something, re-updating PR

datapythonista replied:
I see your point. to me it feels that there is the class `GroupBy`, but here it's not talking about the class, but about the concept. I guess we can use `group by` or `groupping by` to disambiguate. But not that important, we can merge it like this and change it later if we think that makes sense.

datapythonista reviewed (commented):
Really nice, not sure about one of the cases, but other than that looks great.

cleconte987 committed:
Update branch
Merge remote-tracking branch 'upstream_main_pandas/master' into changes_to_pandas_remote

cleconte987 committed:
Remove exception "GroupBy" and lower "GroupBy" in doc

cleconte987 commented:
Ok.. again failures in the PR request sorry for that. I know now why I putted "GroupBy" as an exception. Because it is written like that in title in file doc/source/reference/groupby.rst
I will re-update the PR. Should I add it to exceptions or not? I think it should

cleconte987 committed:
modifying for PR

cleconte987 committed:
Merge remote-tracking branch 'upstream_main_pandas/master' into changes_to_pandas_remote

cleconte987 committed:
Update "groupby" to "GroupBy"

cleconte987 committed:
Merge remote-tracking branch 'upstream_main_pandas/master' into changes_to_pandas_remote

datapythonista reviewed (approved):
Thanks, nice work.

cleconte987 commented:
Thank you.

Yes I understand now for the case of "groupby/GroupBy", but yeah it was not passing the tests then. Ok :)

simonjayhawkins commented:
@cleconte987 can you merge upstream master to resolve conflicts

cleconte987 committed:
merge upstream

datapythonista commented:
I guess everything that this PR was proposing has already been fixed in master, since no changes appear in the diff. Closing, let me know if that's not correct @cleconte987, happy to reopen this is the problem is another one.

cleconte987 commented:
No everything was updated, indeed!
